HW2/P2/P2.py

```diff
@@ -22,30 +22,52 @@
     total = orig_counts.sum()
 
     # serial move
-    counts = orig_counts.copy()
-    with Timer() as t:
-        move_data_serial(counts, src, dest, 100)
-    assert counts.sum() == total, "Wrong total after move_data_serial"
-    print("Serial uncorrelated: {} seconds".format(t.interval))
+    times = []
+    for i in range(1):
+        counts = orig_counts.copy()
+        with Timer() as t:
+            move_data_serial(counts, src, dest, 100)
+        assert counts.sum() == total, "Wrong total after move_data_serial"
+        
+        current_time = t.interval
+        print("Serial uncorrelated: {} seconds".format(current_time))
+        times.append(current_time)
+
+    print "Average", np.mean(times), "\n\n"
+
     serial_counts = counts.copy()
 
     # fine grained
-    counts[:] = orig_counts
-    with Timer() as t:
-        move_data_fine_grained(counts, src, dest, 100)
-    assert counts.sum() == total, "Wrong total after move_data_fine_grained"
-    print("Fine grained uncorrelated: {} seconds".format(t.interval))
+    times = []
+    for i in range(1):
+        counts[:] = orig_counts
+        with Timer() as t:
+            move_data_fine_grained(counts, src, dest, 100)
+        assert counts.sum() == total, "Wrong total after move_data_fine_grained"
+
+        current_time = t.interval
+        print("Fine grained uncorrelated: {} seconds".format(current_time))
+        times.append(current_time)
+
+    print "Average", np.mean(times), "\n\n"
 
     ########################################
     # You should explore different values for the number of locks in the medium
     # grained locking
     ########################################
     N = 10
-    counts[:] = orig_counts
-    with Timer() as t:
-        move_data_medium_grained(counts, src, dest, 100, N)
-    assert counts.sum() == total, "Wrong total after move_data_medium_grained"
-    print("Medium grained uncorrelated: {} seconds".format(t.interval))
+    times = []
+    for N in [1,5,10,20,50,100]:
+        counts[:] = orig_counts
+        with Timer() as t:
+            move_data_medium_grained(counts, src, dest, 100, N)
+        assert counts.sum() == total, "Wrong total after move_data_medium_grained"
+
+        current_time = t.interval
+        print("Medium grained uncorrelated with {} locks: {} seconds".format(N, current_time))
+        times.append(current_time)
+
```

Comment:
I like how you have it spit out exactly the result for the different N's. It would not be that much harder to plot from here and I think it could greatly improve your insight as to what is faster.